### DISCUSSION TITLE
Incrementally Adopting Next.js - component by component

### DISCUSSION THREAD

**Discussion by klnwsk** [vercel/next.js#40222] [Help]
I'm trying to migrate big project which is based on Django framework. Unfortunately I cannot follow the path of migrating view by view because they are complex and I wanted to do it component by component until I have majority of specific view handled by Next.js and then make full switch.

I found some comments and issues stating that Next.js is meant to render full pages rather than components embeded to already existing website but those information are from 2020 and I was wondering if there is a way to handle this?

To sum up: I have website written in Django, I need to keep Django as long as I can until full path can be migrated to Next.js. Until that I want to use Next.js to render parts of the current view. How to achieve that?

I was able to achieve that by creating custom node server which would render React components and then inject them in Django template (HTML) and handle hydration but it's not event close to what Next.js server has to offer.
